skia/RSAsyncStorageModule.cpp

```diff
@@ -28,49 +71,162 @@ auto RSAsyncStorageModule::getMethods() -> std::vector<Method> {
   return {
       Method(
           "multiGet",
-          [] (dynamic args, CxxModule::Callback cb){
-            RNS_LOG_NOT_IMPL;
-            cb({});
-            return;
+          [this] (dynamic args, CxxModule::Callback cb){
+            multiGet(args, cb);
           }),
       Method(
           "multiSet",
-          [] (dynamic args, CxxModule::Callback cb) {
-            RNS_LOG_NOT_IMPL;
-            cb({});
-            return;
+          [this] (dynamic args, CxxModule::Callback cb) {
+            multiSet(args, cb);
           }),
       Method(
           "multiRemove",
-          [] (dynamic args, CxxModule::Callback cb) {
-            RNS_LOG_NOT_IMPL;
-            cb({});
-            return;
+          [this] (dynamic args, CxxModule::Callback cb) {
+            multiRemove(args, cb);
           }),
       Method(
           "mergeItem",
-          [] (dynamic args, CxxModule::Callback cb) {
-            RNS_LOG_NOT_IMPL;
-            cb({});
-            return;
+          [this] (dynamic args, CxxModule::Callback cb) {
+            mergeItem(args, cb);
           }),
       Method(
           "getAllKeys",
-          [] (dynamic args, CxxModule::Callback cb) {
-            RNS_LOG_NOT_IMPL;
-            cb({});
-            return;
+          [this] (dynamic args, CxxModule::Callback cb) {
+            getAllKeys(args, cb);
           }),
       Method(
           "clear",
-          [] (dynamic args, CxxModule::Callback cb) {
-            RNS_LOG_NOT_IMPL;
-            cb({});
-            return;
+          [this] (dynamic args, CxxModule::Callback cb) {
+            clear(args, cb);
           }),
 
   };
 }
 
+
+void RSAsyncStorageModule::multiGet(dynamic args, CxxModule::Callback cb) {
+  taskRunner_->dispatch( [this,args,cb](){
+    dynamic errors = folly::dynamic::array;
+    dynamic resultArray = folly::dynamic::array;
+    for (auto& itemArray : args[0]) { 
+      auto pos = appLocalDataFile_.find(itemArray);
+      dynamic errorItem = folly::dynamic::object();
+      if(pos == appLocalDataFile_.items().end()){
+        errorItem["message"] = "Failed to found the get Item";
+        errorItem["key"] = itemArray;
+        errors.push_back(errorItem);
+      }else{
+        dynamic result = folly::dynamic::array;
+        result.push_back(itemArray);
+        result.push_back(appLocalDataFile_[itemArray]);
+        resultArray.push_back(result);
+      }
+    }
+    cb({(errors.empty() ? nullptr : errors), resultArray});
+  });  
+}
+
+void RSAsyncStorageModule::multiSet(dynamic args, CxxModule::Callback cb) {
+  if(!appLocalFile_.is_open()) {
+    dynamic errors = folly::dynamic::object();
+    errors["message"] = "Failed to write the data";
+    errors["key"] = args[0][0].asString();
+    cb({folly::dynamic::array(errors)});
+    return;
```

Comment:
This error needs to be generated for every key and should be inside the dispatch.